latest updates to mainJavascript.js

julianmay9 opened the pull request:
Hey guys,

Everything now works on this file. It's wonderful. We need to figure out whether there are one or two console logs and commented out bits of code that can be deleted too, but otherwise it functions nicely and it's relatively tidy.

Have a look and let me know what you think. Hopefully it looks good to you too!

Julian

Commit:
latest updates to mainJavascript.js

Mosbycr reviewed (changes requested):
need to add city: city, under userName: userName between line 82 and 83. Also on line 13 you have "-virginia" I didn't know if that was a mistake or not. Otherwise looks good. Just update and re-submit please

Commit:
added the city data to Firebase

julianmay9 commented:
Hey everyone, 

I've updated this again so that we now add the city data to the Firebase database.

The state variable only applies when the user inputs "richmond" as it is and it's only relevant for one of our API calls. I changed the value to "-virginia" as it's the simplest way to hold this for the unique situation where we need it. I could delete it entirely but we might get stuff coming through which apply to other Richmonds in the USA. I decided that, since people are likely to test the app with Richmond as one of the most common default values for 'city', it would be best to avoid this.

It's due to a flaw with the APIs which I think our TAs will appreciate. If we had more time, we could probably figure out a better solution.

Julian
